### PR TITLE
Add json schema validation

### PR DESCRIPTION
**Pull request by Thiago-NovaesB**
- hydro.json
- PowerModels.json
- Tests added

### CODE REVIEW

File: src/IO.jl [hunk 1]
```diff
@@ -131,3 +134,40 @@ function create_param(;
...
+    HydroPowerModels.validate_json_powermodels(json_path::String)
+
+Returns nothing if the json 'json_path' matchs the powermodels schema, showns the errors otherwise.
+
+Parameters:
+-   json_path   : Path to json.
+"""
+function validate_json_powermodels(json_path::String)
+    return validate_json(json_path, powermodels_schema_path)
+end
```
Comment (andrewrosemberg) [inline src/IO.jl]:
```suggestion
end

```

File: src/jsonschema/PowerModels.json [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+                        },
+                        "index": {
+                            "type": "integer"
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
```
Comment (andrewrosemberg) [inline src/jsonschema/PowerModels.json:297]:
```suggestion
}

```

File: src/jsonschema/hydro.json [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                    },
+                    "downstream_spill": {
+                        "type": "array",
+                        "items": {}
+                    }
+                }
+            }
+        }
+    }
+}
```
Comment (andrewrosemberg) [inline src/jsonschema/hydro.json:72]:
```suggestion
}

```

File: test/IO.jl [hunk 4]
```diff
@@ -15,4 +15,10 @@ using GLPK
         @test params["optimizer"] == optimizer
         @test params["model_constructor_grid"] == DCPPowerModel
     end
+
+    @testset "Validate Json" begin
+        @test isnothing(validate_json_hydro("..\\testcases\\validate_json\\correct_json.jl"))
+        @test !isnothing(validate_json_hydro("..\\testcases\\validate_json\\missing_json.jl"))
+        @test !isnothing(validate_json_hydro("..\\testcases\\validate_json\\type_json.jl"))
```
Comment (andrewrosemberg) [inline test/IO.jl]:
```suggestion
        @test isnothing(validate_json_hydro(joinpath(testcases_dir, "validate_json", "correct_json.jl")))
        @test !isnothing(validate_json_hydro(joinpath(testcases_dir, "validate_json", "missing_json.jl")))
        @test !isnothing(validate_json_hydro(joinpath(testcases_dir, "validate_json", "type_json.jl")))
```

File: testcases/validate_json/correct_json.jl [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+            "initial_volume":0.18,
+            "production_factor":1,
+            "spill_cost":0,
+            "minimal_outflow_violation_cost":0,
+            "minimal_volume_violation_cost":0,
+            "downstream_turn": [],
+            "downstream_spill": []
+        }
+    ]
+}
```
Comment (andrewrosemberg) [inline testcases/validate_json/correct_json.jl]:
```suggestion
}

```

File: testcases/validate_json/missing_json.jl [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+            "initial_volume":0.18,
+            "production_factor":1,
+            "spill_cost":0,
+            "minimal_outflow_violation_cost":0,
+            "minimal_volume_violation_cost":0,
+            "downstream_turn": [],
+            "downstream_spill": []
+        }
+    ]
+}
```
Comment (andrewrosemberg) [inline testcases/validate_json/missing_json.jl]:
```suggestion
}

```

File: testcases/validate_json/type_json.jl [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+            "initial_volume":0.18,
+            "production_factor":1,
+            "spill_cost":0,
+            "minimal_outflow_violation_cost":0,
+            "minimal_volume_violation_cost":0,
+            "downstream_turn": [],
+            "downstream_spill": []
+        }
+    ]
+}
```
Comment (andrewrosemberg) [inline testcases/validate_json/type_json.jl]:
```suggestion
}

```

File: Project.toml
Comment (andrewrosemberg) [inline Project.toml:1]: minor or patch version bump needed.

File: src/IO.jl (hunk 1 above)
Comment (Thiago-NovaesB) [reply src/IO.jl]: fix

File: src/jsonschema/PowerModels.json (hunk 2 above)
Comment (Thiago-NovaesB) [reply src/jsonschema/PowerModels.json:297]: necessary in json files?

File: test/IO.jl (hunk 4 above)
Comment (Thiago-NovaesB) [reply test/IO.jl]: fix

File: testcases/validate_json/correct_json.jl (hunk 5 above)
Comment (Thiago-NovaesB) [reply testcases/validate_json/correct_json.jl]: It's a json now

File: testcases/validate_json/missing_json.jl (hunk 6 above)
Comment (Thiago-NovaesB) [reply testcases/validate_json/missing_json.jl]: It's a json now

File: testcases/validate_json/type_json.jl (hunk 7 above)
Comment (Thiago-NovaesB) [reply testcases/validate_json/type_json.jl]: It's a json now

File: Project.toml
Comment (Thiago-NovaesB) [reply Project.toml:1]: fix

File: src/jsonschema/hydro.json (hunk 3 above)
Comment (Thiago-NovaesB) [reply src/jsonschema/hydro.json:72]: necessary in json files?

File: src/jsonschema/PowerModels.json (hunk 2 above)
Comment (andrewrosemberg) [reply src/jsonschema/PowerModels.json:297]: You are right!

File: src/jsonschema/hydro.json (hunk 3 above)
Comment (andrewrosemberg) [reply src/jsonschema/hydro.json:72]: I guess not. You are right

**Review by andrewrosemberg — APPROVED** [review]
LGTM

**Pull request merged**